packages/obonode/obojobo-sections-assessment/viewer.js

```diff
@@ -33,6 +33,14 @@ Common.Registry.registerModel('ObojoboDraft.Sections.Assessment', {
 
 			return AssessmentUtil.getAttemptsRemaining(viewerProps.assessmentState, assessmentModel)
 		},
+		'assessment:attemptsTaken'(textModel, viewerProps) {
+			const assessmentModel = textModel.getParentOfType('ObojoboDraft.Sessions.Assessment')
+			if (!assessmentModel) {
+				return null
+			}
+
+			return viewerProps.assessmentState.attempts.length
```

Comment:
Some backstory on this - The various states on viewerProps, like assessmentState, usually should be treated as a black box. Instead there's various "classes" which help you interpret this data. So instead of digging in `assessmentState` you can pass the state to `AssessmentUtil`, and AssessmentUtil has `getNumberOfAttemptsCompletedForModel`.

So this line should become
```javascript
return AssessmentUtil.getNumberOfAttemptsCompletedForModel(
	viewerProps.assessmentState,
	assessmentModel
)
```